### PR TITLE
Add Make target for building and pushing images

### PR DESCRIPTION
**Pull request by RadekManak**
Adds targets to Makefile for building the images and pushing to registry. These targets are useful when debugging. It allows you to test your changes in a running cluster by pushing your development images to your quay repository and have cluster deploy them by pointing the CPMS deployment to it.

### CODE REVIEW

**Review by JoelSpeed — COMMENTED** [review]
Seems useful, have you tested that you can push the images using both docker and podman?

**Comment by RadekManak** [comment]
I have not 😞. Looks like docker does not have the option to specify where to push the image. Instead, you have to specify it in the name of the built image. Podman can also work like this. I've dropped the PUSH_REPO variable for the sake of compatibility

File: Makefile [hunk 1]
```diff
@@ -20,6 +20,22 @@ ENVTEST = go run ${PROJECT_DIR}/vendor/sigs.k8s.io/controller-runtime/tools/setu
 GINKGO = go run ${PROJECT_DIR}/vendor/github.com/onsi/ginkgo/v2/ginkgo
 GOLANGCI_LINT = go run ${PROJECT_DIR}/vendor/github.com/golangci/golangci-lint/cmd/golangci-lint
 
+VERSION     ?= $(shell git describe --always --abbrev=7)
+MUTABLE_TAG ?= latest
+IMAGE        = cluster-control-plane-machine-set-operator
```
Comment (JoelSpeed) [inline Makefile:25]: Is this overridable?

Comment (RadekManak) [reply Makefile]: Fixed, Now it can be overridden by environment variable.

**Review by JoelSpeed — COMMENTED** [review]
/approve
/lgtm

**Comment by JoelSpeed** [comment]
/override ci/prow/e2e-aws-upgrade

This is a tooling change, no effect to the payload

**Comment by JoelSpeed** [comment]
/override ci/prow/e2e-aws-upgrade

This is a tooling change, no effect to the payload

**Comment by JoelSpeed** [comment]
/override ci/prow/e2e-aws-upgrade

This is a tooling change, no effect to the payload

**Comment by JoelSpeed** [comment]
/override ci/prow/e2e-aws-serial

This is a tooling change, no effect to the payload

**Pull request merged**